### PR TITLE
Add doc comments for WASI FD associated constants

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
I want to try out something I'm going to call "daily docs" where I write documentation for one thing every day

### CODE REVIEW

File: lib/wasi/src/state/mod.rs [hunk 1]
```diff
@@ -128,10 +128,18 @@ pub struct Fd {
 }
 
 impl Fd {
+    /// This FD can be used with read system calls.
```
Comment (nlewycky) [inline lib/wasi/src/state/mod.rs]: FD is unexplained, you could either write "file descriptor" here, or "descriptor" or document "pub struct Fd" above?

File: lib/wasi/src/state/mod.rs [hunk 2]
```diff
@@ -128,10 +128,18 @@ pub struct Fd {
...
     pub const READ: u16 = 1;
+    /// This FD can be used with write system calls.
     pub const WRITE: u16 = 2;
+    /// This FD can append in write system calls. Note that append
```
Comment (nlewycky) [inline lib/wasi/src/state/mod.rs]: "Note that append permissions require the write permission."

What happens if I open for read and append but not write? Does the open fail? Does it automatically set write for me?

File: lib/wasi/src/state/mod.rs [hunk 3]
```diff
@@ -128,10 +128,18 @@ pub struct Fd {
 }
 
 impl Fd {
```
Comment (nlewycky) [inline lib/wasi/src/state/mod.rs]: Are there any invariants across the whole group? Like, all file descriptors must be open for at least reading or writing?

File: lib/wasi/src/state/mod.rs (hunk 1 above)
Comment (MarkMcCaskey) [reply lib/wasi/src/state/mod.rs]: Yeah that should be ```[`Fd`]```, good call

**Pull request merged**